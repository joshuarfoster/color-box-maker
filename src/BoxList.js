import React, { useState } from "react";
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";

function BoxList() {
    const [boxes, setBoxes] = useState([])
    const addBox = (width, height, color) => {
        setBoxes(boxes => [...boxes, {width,height,color}])
    }
    const deleteBox = (id) => {
        const newBoxes = boxes.filter((box, idx)=> idx !== id)
        setBoxes(newBoxes)
    }

    return (
        <div>
            <h3>Create New Box</h3>
            <NewBoxForm addBox={addBox} />
            <div>
                {boxes.map((box, idx) => <Box width={box.width} height={box.height} color={box.color} idx={idx} deleteBox={deleteBox}/>)}
            </div>
        </div>
    )
}

export default BoxList