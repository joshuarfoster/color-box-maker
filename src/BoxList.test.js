import React from "react";
import { render, fireEvent} from "@testing-library/react";
import BoxList from "./BoxList";

function addBox (boxList, height = '100px', width = '100px', color = "#ffffff") {
    const heightInput = boxList.getByLabelText(/Height/i);
    const widthInput = boxList.getByLabelText(/Width/i);
    const colorInput = boxList.getByLabelText(/Color/i);
    fireEvent.change(colorInput, { target: { value: color } });
    fireEvent.change(widthInput, { target: { value: width } });
    fireEvent.change(heightInput, { target: { value: height } });
    const button = boxList.queryByRole("button", { type: "submit" });
    fireEvent.click(button);
};

it("renders without crashing", function () {
    render(<BoxList/>);
});

it("matches snapshot", function () {
    const {asFragment} = render(<BoxList/>);
    expect(asFragment()).toMatchSnapshot();
});

it("should add new box", function () {
    const boxList = render(<BoxList/>);
    addBox(boxList);
    const deleteBtn = boxList.queryByText("X")
    expect(deleteBtn).toBeInTheDocument();
});

it("should delete old box", function () {
    const boxList = render(<BoxList/>);
    addBox(boxList);
    const deleteBtn = boxList.queryByText("X");
    fireEvent.click(deleteBtn);
    expect(deleteBtn).not.toBeInTheDocument();
});