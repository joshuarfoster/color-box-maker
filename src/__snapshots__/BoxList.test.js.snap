// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Create New Box
    </h3>
    <form>
      <label
        for="width"
      >
        Width : 100
      </label>
      <input
        id="width"
        max="200"
        min="0"
        name="width"
        type="range"
        value="100"
      />
      <label
        for="height"
      >
        Height : 100
      </label>
      <input
        id="height"
        max="200"
        min="0"
        name="height"
        type="range"
        value="100"
      />
      <label
        for="color"
      >
        Color : 
      </label>
      <input
        id="color"
        name="color"
        type="color"
        value="#ffffff"
      />
      <input
        type="submit"
      />
    </form>
    <div />
  </div>
</DocumentFragment>
`;
